### PR TITLE
Improvements to the `sda-admin` script

### LINKED ISSUE #35: sda-admin behaves differently when ingesting folders depending on trailing `/`
Given a correctly encrypted file `test_file.c4gh` but uploaded to a target folder, <folder>, without its *.c4gh extension `sda-admin` fails to ingest as follows:

```
s3cmd -c s3cmd.conf put test_file.c4gh s3://jd123_lifescience-ri.eu/bams/test2_file.bam
```

These fail:
```
./scripts/sda-admin ingest bams
./scripts/sda-admin ingest bams/test2_file.bam
```

This works:
```
./scripts/sda-admin ingest bams/
```

What makes the difference is the trailing `/`.

### LINKED ISSUE #37: Fix various issues in the `sda-admin` script
After seeing some initial use, the `sda-script` now needs to be adjusted in light of user feedback:

- [x] The script should ideally not make any assumptions about filenames.
   - When _uploading_ files, we will still re-encrypt files, adding the `.c4gh` filenames suffix, but we will make no assumptions about the names of already uploaded files.

- [x] The script should not assume that the `curl` utility has the `--json` option (available in newer releases).
    - Since `--json` is just a convenience option, it's easy to replace it with the more portable set of options that it aliases.
 
- [x] The script should be able to work with any storage bucket, not just the bucket belonging to the user in the configuration file.
   - The username is already settable on the command line using `--s3-access-key`, ~but it also makes sense to be able to use an _empty_ access key when listing things, which then would show everything (not just the contents of the bucket configured in the configuration file).~
   - Add an `-a` option to the relevant operations that one could use to "list all". 

- [x] The `--s3-access-key` option is unintuitive (it overrides the `access_key` configuration option in the `s3cmd` configuration file).
   - We may want an `--user` option that aliases this.

- [x] The username must be sanity-checked.
   - Any `@` character should be replaced by a `_` character.

### PR DESCRIPTION
**Pull request by andkaha**
This PR addresses the issues in #37  

This PR closes #35
This PR closes #37

### CODE REVIEW

**Review by jbygdell — APPROVED** [review]

File: scripts/sda-admin [hunk 1]
```diff
@@ -21,8 +21,9 @@ SDA_KEY=${SDA_KEY-crypt4gh_key.pub}			# --sda-key pathname
 # not set, the tool will be picked up from "$PATH" as usual.
 SDA_CLI=${SDA_CLI-sda-cli}	# --sda-cli pathname
 
-# There is also S3_ACCESS_KEY (--s3-access-key), but since its value
-# depend on the final value of "$SDA_CONFIG", we can't set it here.
+# There is also S3_ACCESS_KEY (--s3-access-key and its alias, --user),
+# but since its value depend on the final value of "$SDA_CONFIG", we
```
Comment (aaperis) [inline scripts/sda-admin]:
```suggestion
# but since its value depends on the final value of "$SDA_CONFIG", we
```

**Review by aaperis — COMMENTED** [review]
This looks good! I made a few tests and I have some notes:
1. the script does not return any error message if I provide a wrong path to a file that I want to upload, i.e. if the file doesnot exist
2. I am skeptical about the `""` option for `sda-admin accession` and `sda-admin dataset` since this could lead to unwanted situations. 
Consider for example that more than one user have uploaded files and then the admin accidentally does a mass accession or dataset assignment using `""`. So, I would suggest that we keep the `""` option only for the `ingest` subcommand.

**Review by kostas-kou — APPROVED** [review]
Looks very good and I have tested it a bit around. I have just a minor comment which is maybe not so relevant with this PR but it would be nice if there was the `-k` option for insecure connections in the curl command. I am making this comment because I am using this script for the Big Picture and it is not working without the insecure connection.

**Comment by andkaha** [comment]
> Looks very good and I have tested it a bit around. I have just a minor comment which is maybe not so relevant with this PR but it would be nice if there was the `-k` option for insecure connections in the curl command. I am making this comment because I am using this script for the Big Picture and it is not working without the insecure connection.

One solution for this that doesn't require that you change the script:
```
echo '--insecure' >.curlrc
export CURL_HOME=$PWD
```

Basically, this creates a Curl config file in the current directory containing the `--insecure` option. By setting `CURL_HOME` to the path of the current directory, you make all invocations of `curl` read that file first.

**Comment by kostas-kou** [comment]
> export CURL_HOME=$PWD

It works like a charm. Thanks Andreas

**Comment by andkaha** [comment]
> This looks good! I made a few tests and I have some notes:
> 
> 1. the script does not return any error message if I provide a wrong path to a file that I want to upload, i.e. if the file doesnot exist
> 2. I am skeptical about the `""` option for `sda-admin accession` and `sda-admin dataset` since this could lead to unwanted situations.
>    Consider for example that more than one user have uploaded files and then the admin accidentally does a mass accession or dataset assignment using `""`. So, I would suggest that we keep the `""` option only for the `ingest` subcommand.

These comments hopefully addressed with the most recent commit.

**Review by aaperis — APPROVED** [review]
Looks great!

**Pull request merged**